packages/flutter/test/material/text_field_test.dart

```diff
@@ -6087,6 +6087,43 @@ void main() {
       expect(find.byType(CupertinoButton), findsNothing);
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
 
+  testWidgets(
+    'tap with a mouse does not move cursor to the edge of the word',
+    (WidgetTester tester) async {
+      final TextEditingController controller = TextEditingController(
+        text: 'Atwater Peel Sherbrooke Bonaventure',
+      );
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Material(
+            child: Center(
+              child: TextField(
+                controller: controller,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      final Offset textfieldStart = tester.getTopLeft(find.byType(TextField));
+
+      final TestGesture gesture = await tester.startGesture(
+        textfieldStart + const Offset(50.0, 9.0),
+        pointer: 1,
+        kind: PointerDeviceKind.mouse,
+      );
+      addTearDown(gesture.removePointer);
+      await tester.pump();
```

Comment:
Probably because mouse region takes a frame after `addPointer` to update itself? If so I recommend doing it in the following order:
- createGesture
- addPointer
- pumpWidget
- down
- up (or other gestures)

Reply:
Perhaps it is to allow the focus to take effect.  That also happens in a microtask.

Reply:
we should figure out why we need the pump and say so in a comment

Reply:
I'll figure out if the pump is needed or not, and either remove it or explain why it's needed.